build/commands/scripts/sync.js

```diff
@@ -41,6 +41,10 @@ program
     '--with_issue_44921',
     'Do not pass --revision to gclient to avoid process hanging on jenkins. https://github.com/brave/brave-browser/issues/44921',
   )
+  .option(
+    '--checkout_clangd',
```

Comment:
instead consider pulling this bool from `.env`. no need to type it each time.

Reply:
Yeah, ideally we shouldn't have to add a config entry for each of them, but we could have a custom_vars env with a list that gets appended?

Reply:
probably use same format as gn/ninja with `:` separator

Reply:
But it isn't just that `custom_vars` gets wiped, it's also that `target_os` (and anything else) is too. IMO ideally we have either one of the two solutions:
- they can all come from `.env` if we are admitting that `.gclient` is generated and shouldn't be manually edited 
- _or_ we decide that `.gclient` is the user's responsibility after initial creation and we don't modify

I can understand if we want to be able to re-create .gclient if we need change it for some reason like adding a variable to all brave developers. So, perhaps `.env` is the way to go?

For the record, we only re-create `.gclient` when either `--init`, `--target_os` or `--target_arch` are passed to the sync command. I don't think there's any other case.